Kodi 18.6 segfaults playing mpeg2 video content on Ubuntu

## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Using 2:18.6+git20200301.1506-final-0eoan (Ubuntu 19.10 Eoan Ermine backport).

Kodi crashes/segfaults whenever I try to play mpeg2-encoding video content, regardless of whether the container is m2ts, mkv, or vob.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Select video content with mpeg2 encoding. It should play.


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Kodi segfaults and crashes out instead of playing the content.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
Unknown


### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. Launch kodi 
2. Select content with mpeg2-encoded video content


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:
[kodi-crash-log-mpeg2-m2ts-content.txt](https://github.com/xbmc/xbmc/files/4462990/kodi-crash-log-mpeg2-m2ts-content.txt)
[kodi-crash-log-mpeg2-vob-content.txt](https://github.com/xbmc/xbmc/files/4462991/kodi-crash-log-mpeg2-vob-content.txt)


### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->
No screenshots illustrate problem, since it just crashes.


## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->
No additional screenshots.


### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Ubuntu 19.10 Eoan Ermine
 - Kodi version: 18.6



<!--- End of this issue -->

Upgraded MESA to:
```
2020-04-10 15:55:16.567 T:140006575642112  NOTICE: GL_RENDERER = AMD CEDAR (DRM 2.50.0 / 5.3.0-46-generic, LLVM 10.0.0)
2020-04-10 15:55:16.567 T:140006575642112  NOTICE: GL_VERSION = 3.3 (Core Profile) Mesa 20.0.3 - kisak-mesa PPA
```
Same segfault on playing mpeg2 content.

I tried to build git master branch but ran into trouble:
```src/xbmc/CMakeFiles/CheckIncludeFiles/HAVE_LINUX_DMA_HEAP.c:2:10: fatal error: linux/dma-heap.h: No such file or directory```

Seems like I need a newer kernel?

I also tried building 18.6 from git (18.6 Git:20200408-6b7f5c7f0e). This built fine but I get the same segfault when selecting mpeg2 content, even with MESA 20.0.3.

> Upgraded MESA to:
> 
> ```
> 2020-04-10 15:55:16.567 T:140006575642112  NOTICE: GL_RENDERER = AMD CEDAR (DRM 2.50.0 / 5.3.0-46-generic, LLVM 10.0.0)
> 2020-04-10 15:55:16.567 T:140006575642112  NOTICE: GL_VERSION = 3.3 (Core Profile) Mesa 20.0.3 - kisak-mesa PPA
> ```
> 
> Same segfault on playing mpeg2 content.
> 

I would make a bug report at the mesa repo


> I tried to build git master branch but ran into trouble:
> `src/xbmc/CMakeFiles/CheckIncludeFiles/HAVE_LINUX_DMA_HEAP.c:2:10: fatal error: linux/dma-heap.h: No such file or directory`
> 
> Seems like I need a newer kernel?

This should not be a fatal error, That log line doesn't look complete to me.

--- a/system/settings/linux.xml
+++ b/system/settings/linux.xml
@@ -26,7 +26,7 @@
           <requirement>HAVE_LIBVDPAU</requirement>
           <visible>false</visible>
           <level>3</level>
-          <default>true</default>
+          <default>false</default>
           <dependencies>
             <dependency type="enable">
               <condition setting="videoplayer.usevdpau" operator="is">true</condition> <!-- USE VDPAU -->
@@ -74,7 +74,7 @@
             </dependency>
           </dependencies>
           <level>3</level>
-          <default>true</default>
+          <default>false</default>
           <control type="toggle" />
         </setting>
         <setting id="videoplayer.usevaapimpeg4" type="boolean" parent="videoplayer.usevaapi" label="13449" help="13450">

